utils/src/ooinstall/openshift_ansible.py

```diff
@@ -175,30 +184,20 @@ def write_proxy_settings(base_inventory):
         pass
 
 
-# pylint: disable=too-many-branches
 def write_host(host, role, inventory, schedulable=None):
     global CFG
 
     if host.preconfigured:
         return
 
     facts = ''
-    if host.ip:
-        facts += ' openshift_ip={}'.format(host.ip)
-    if host.public_ip:
-        facts += ' openshift_public_ip={}'.format(host.public_ip)
-    if host.hostname:
-        facts += ' openshift_hostname={}'.format(host.hostname)
-    if host.public_hostname:
-        facts += ' openshift_public_hostname={}'.format(host.public_hostname)
-    if host.containerized:
-        facts += ' containerized={}'.format(host.containerized)
+    for prop in HOST_VARIABLES_MAP:
+        if getattr(host, prop):
+            facts += ' {}={}'.format(HOST_VARIABLES_MAP.get(prop), getattr(host, prop))
+
     if host.other_variables:
         for variable, value in host.other_variables.iteritems():
             facts += " {}={}".format(variable, value)
-    if host.node_labels:
-        if role == 'node':
-            facts += ' openshift_node_labels="{}"'.format(host.node_labels)
```

Comment:
I made a simple patch for this. Built the rpm and installed it. Now I get correct non-exploding output.

~~We may want to consider adding a unit test case for this in the future. The work I did refactoring before included a new unittest to verify current vs. my new method output for writing the hosts file. It actually does 99% of the work to implement a unittest to verify things like this. ~~

I added a unittest to my fork of smunilla/BZ1368414 

* https://github.com/tbielawa/openshift-ansible/commit/2733b8f9cfaca3a4b1c73a5e9df6f757b8581d99 is the actual code fix
* https://github.com/tbielawa/openshift-ansible/commit/a41d7232d665bd126b6ea525e0869c5fd28ef33e adds a unittest that verifies the node labels are written out in double quotes.